### PR TITLE
Add PodAnnotations feature

### PR DESCRIPTION
**Pull request by JonathanLachapelle**
## Proposed changes

Pod annotations are not available for all deployment with splunk-connect, I added the possibility to set a pod annotations for all daemon set and deployment. This is very usefull if you have for instance istio and you want to disable the istio side-car injection.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](https://github.com/splunk/splunk-connect-for-kubernetes/blob/develop/CONTRIBUTING.md) doc
- [X] I have read the [CLA](https://github.com/splunk/splunk-connect-for-kubernetes/blob/develop/CLA.md)
- [X] I have added necessary documentation (if appropriate)
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by JonathanLachapelle** [comment]
Anyone can re-run the build_test, it's not related to my changes.

**Comment by rockb1017** [comment]
re-running. I had to enable sharing environment variable. I recently added it for ruby version.

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-objects/values.yaml [hunk 1]
```diff
@@ -189,7 +189,10 @@ image:
 
 # Environment variable for metrics daemonset
 environmentVar:
-    
+
+# Pod annotations for metrics daemonset
```
Comment (rockb1017) [inline helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-objects/values.yaml:193]: metrics -> objects

**Review by rockb1017 — CHANGES_REQUESTED** [review]
Could you also update the values.yaml file in the umbrella chart? `helm-chart/splunk-connect-for-kubernetes/values.yaml`

**Pull request merged**